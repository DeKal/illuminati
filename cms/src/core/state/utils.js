export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
