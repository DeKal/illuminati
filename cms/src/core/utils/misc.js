import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url, isExact = false) => {
  if (url === '/' || pathName === '/' || isExact) {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}

export const arrayToAttributes = array =>
  array.reduce((object, attribute) => {
    return {
      ...object,
      [attribute]: true
    }
  }, {})

export const attributesToArray = obj =>
  Object.keys(obj).map(attribute => (obj[attribute] ? attribute : null))
