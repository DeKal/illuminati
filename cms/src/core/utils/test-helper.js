import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
