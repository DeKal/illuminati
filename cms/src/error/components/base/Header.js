import React from 'react'
import { styled } from '@material-ui/core/styles'
import PropTypes from 'prop-types'

const Header = ({ children }) => {
  return (
    <Container>
      <Style>{children}</Style>
    </Container>
  )
}

Header.propTypes = {
  children: PropTypes.string
}

export default Header

const Container = styled('div')({
  height: '280px',
  position: 'relative',
  zIndex: -1,

  '@media only screen and (max-width: 767px)': {
    height: '142px'
  }
})

const Style = styled('h1')({
  fontSize: '200px',
  margin: 0,
  fontWeight: 900,
  position: 'absolute',
  left: '50%',
  '-webkit-transform': 'translateX(-50%)',
  '-ms-transform': 'translateX(-50%)',
  transform: 'translateX(-50%)',
  background: 'url(/images/oops.jpg) no-repeat',
  '-webkit-background-clip': 'text',
  '-webkit-text-fill-color': 'transparent',
  backgroundSize: 'cover',
  backgroundPosition: 'center',

  '@media only screen and (max-width: 767px)': {
    fontSize: '100px'
  }
})
