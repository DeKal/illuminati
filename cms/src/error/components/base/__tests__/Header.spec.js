import * as React from 'react'
import { mountWithTheme } from 'core/utils/test-helper'
import Header from '../Header'

describe('Header', () => {
  it('should render correctly', function() {
    const wrap = mountWithTheme(<Header />)
    expect(wrap).toMatchSnapshot()
  })
})
