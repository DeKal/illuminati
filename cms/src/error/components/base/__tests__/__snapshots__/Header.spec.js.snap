// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <Header>
    <Styled(div)>
      <div
        className="div-root-1"
      >
        <Styled(h1)>
          <h1
            className="h1-root-2"
          />
        </Styled(h1)>
      </div>
    </Styled(div)>
  </Header>
</div>
`;
